app/src/main/java/com/cornellappdev/uplift/ui/components/onboarding/PhotoPicker.kt

```diff
@@ -0,0 +1,131 @@
+import androidx.compose.foundation.Image
+import androidx.compose.foundation.background
+import androidx.compose.foundation.layout.Arrangement
+import androidx.compose.foundation.layout.Box
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.foundation.layout.offset
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.layout.size
+import androidx.compose.foundation.shape.CircleShape
+import androidx.compose.material3.CardDefaults
+import androidx.compose.material3.ElevatedCard
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.saveable.rememberSaveable
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.draw.clip
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.layout.ContentScale
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import coil.compose.rememberAsyncImagePainter
+import com.cornellappdev.uplift.util.GRAY01
+import com.cornellappdev.uplift.R
+
+/***
+ * Component that displays a camera icon for the user to select a photo for their profile
+ *  @param onPhotoSelected: function to call when the user selects a photo. Takes in uri parameter
+ */
+@Composable
+fun PhotoPicker(onPhotoSelected: (Uri) -> Unit){
+    // State to store the selected image URI
+    var selectedImageUri by rememberSaveable { mutableStateOf<Uri?>(null) }
+
+    // Registers a photo picker activity launcher in single-select mode.
+    val pickMedia = rememberLauncherForActivityResult(ActivityResultContracts.PickVisualMedia()) { uri ->
+        // Callback is invoked after the user selects a media item or closes the
+        // photo picker.
+        if (uri != null) {
+            selectedImageUri = uri
+            onPhotoSelected(uri)
+        }
+    }
+
+    Box(
+        modifier = Modifier
+            .size(160.dp),
+        contentAlignment = Alignment.Center
+    ){
+        ElevatedCard(
+            elevation = CardDefaults.cardElevation(
+                defaultElevation = 10.dp
+            ),
+            colors = CardDefaults.cardColors(
+                containerColor = Color.White,
+            ),
+            shape = CircleShape,
+            modifier = Modifier
+                .size(160.dp),
+            onClick = {
+                // Launch the photo picker and let the user choose only images.
+                pickMedia.launch(PickVisualMediaRequest(ActivityResultContracts.PickVisualMedia.ImageOnly))
+            }
+        ){
+            Box(
+                modifier = Modifier
+                    .fillMaxSize(),
+                contentAlignment = Alignment.Center
+            ){
```

Comment:
Any reason for *this* box? It doesnt seem to overlay anything in particular on top of anything.

Reply:
Yeah, I wasn't sure if there was a better way to go about it, but I couldn't seem to get the grey circle with camera icon to be centered without it, but I'm sure there is probably a better way to go about it. Will probably switch to the whole frame from figma, so this might not apply afterwards.